src/CryptoFraudDetection/scraper/twitter.py

```diff
@@ -7,3 +7,304 @@
+            )
+            driver.execute_script("arguments[0].click();", close_button)
+            print("Close button clicked successfully.")
+        except Exception as e:
+            self.logger.warning("Could not find or click the close button:", e)
+
+
+    def perform_search(self, driver: webdriver.Firefox, search_query: str) -> None:
+        """
+        Performs a search for the given query.
+
+        Args:
+            driver (webdriver.Firefox): Selenium WebDriver instance.
+            search_query (str): The search query to look for.
+        """
+        try:
+            search_bar = WebDriverWait(driver, 10).until(
+                EC.element_to_be_clickable((By.XPATH, "//input[@aria-label='Search query']"))
+            )
+            search_bar.clear()  # Clear the search bar before typing
+            search_bar.send_keys(search_query)
+            search_bar.send_keys(Keys.RETURN)  # Press Enter key to submit the search
+            print(f"Searched for: {search_query}")
+            time.sleep(5)
+        except Exception as e:
+            self.logger.error("Could not find or enter into the search bar:", e)
+
+
+    def scrape_tweets(self, driver: webdriver.Firefox, tweet_count: int, search_query: str) -> pd.DataFrame:
+        """
+        Scrapes the specified number of tweets and returns a DataFrame.
+
+        Args:
+            driver (webdriver.Firefox): Selenium WebDriver instance.
+            tweet_count (int): Number of tweets to scrape.
+            search_query (str): The search query used.
+
+        Returns:
+            pd.DataFrame: DataFrame containing tweet data.
+        """
+        tweets_scraped = 0
+        tweet_data = []
+
+        while tweets_scraped < tweet_count:
+            tweets = self.get_tweets(driver)
+            for tweet in tweets:
+                try:
+                    username, content, timestamp, likes, impressions = self.extract_tweet_details(tweet)
+                    tweet_data.append([username, content, timestamp, likes, impressions])
+                    tweets_scraped += 1
+                    print(f"Scraped tweet {tweets_scraped}/{tweet_count}: {username} - {content[:50]}")
+
+                    if tweets_scraped >= tweet_count:
+                        break
+
+                    sleep_time = random.uniform(2, 5)
+                    print(f"Sleeping for {sleep_time:.2f} seconds...")
+                    time.sleep(sleep_time)
+
+                except Exception as e:
+                    self.logger.warning(f"Could not extract tweet details: {e}")
+
+            driver.execute_script("window.scrollTo(0, document.body.scrollHeight);")
+            time.sleep(3)
+
+        # Convert to DataFrame
+        df = pd.DataFrame(tweet_data, columns=["Username", "Tweet", "Timestamp", "Likes", "Impressions"])
+        print("Tweets scraped into DataFrame.")
+        return df
```

Comment:
sorry for telling you this late, but could you change it so that it returns a dictionary in this format {"column1": {"value1", "value2", "value3"}, "column2": {"value1", "value2", "value3"}}

Reply:
this way it's the same as in the other scrapers